consensus/core/src/synchronizer.rs

```diff
@@ -418,20 +673,33 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher> Synchronizer<C
         }
 
         let mut peers = peers.into_iter();
-
         let mut request_futures = FuturesUnordered::new();
 
+        // get the highest accepted rounds
+        let highest_rounds = match core_dispatcher.get_highest_accepted_rounds().await {
+            Ok(rounds) => rounds,
+            Err(err) => {
+                debug!("Core is shutting down, synchronizer is shutting down: {err:?}");
+                return vec![];
+            }
+        };
+
         // Send the initial requests
         for blocks in missing_blocks.chunks(MAX_FETCH_BLOCKS_PER_REQUEST) {
             let peer = peers
                 .next()
                 .expect("Possible misconfiguration as a peer should be found");
             let block_refs = blocks.iter().cloned().collect::<BTreeSet<_>>();
 
+            // lock the blocks to be fetched. Allow re-fetching from same peer
+            let blocks_guard = inflight_blocks
+                .lock_blocks(&context, block_refs.clone(), peer, true)
```

Comment:
Pay attention that here we are calling the `lock_blocks` with `skip_checks = true` . I gave thought around this and it would be simpler to just always trying to acquire a lock during the scheduler run - as one run of the scheduler runs a t a time - so at worst case scenario we'll have `MAX_AUTHORITIES_TO_FETCH_PER_BLOCK + 1` peers fetching a specific block. Practically we want to guarantee that the scheduler will do the best to fetch the blocks.

Reply:
I think it is ok though not great. Is it complex to avoid having too many concurrent fetches on the same block?